gramps/gen/db/bookmarks.py

```diff
@@ -20,35 +20,50 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 #
 
-#-------------------------------------------------------------------------
+from ..utils.callback import Callback
+
+# -------------------------------------------------------------------------
 #
-# class DbBookmarks
+# DbBookmarks class
 #
-#-------------------------------------------------------------------------
-class DbBookmarks:
-    def __init__(self, default=[]):
-        self.bookmarks = list(default) # want a copy (not an alias)
+# -------------------------------------------------------------------------
+class DbBookmarks(Callback):
+
+    __signals__ = {"bookmark-list-changed": (str,)}
+
+    def __init__(self, default=[], cls_type="Person"):
+        super().__init__()
+        self.cls_type = cls_type
+        self.bookmarks = list(default)  # want a copy (not an alias)
 
-    def set(self, new_list):
+    def set(self, new_list, quiet=False):
         self.bookmarks = list(new_list)
+        if not quiet:
+            self.emit("bookmark-list-changed", (self.cls_type,))
```

Comment:
The `cls_type` isn't necessary here since we know the class type when we connect to the signal.  A common handler could be used as follows:
```
self.db.event_bookmarks.connect('bookmark-list-changed', handler_func, 'Event')
```

Reply:
Yes it could be that way.

Reply:
Cleaned up.